src/c++/perf_analyzer/infer_data_manager_shm.cc

```diff
@@ -384,7 +354,8 @@ InferDataManagerShm::InitInferDataOutput(
 
 cb::Error
 InferDataManagerShm::UpdateInputs(
-    const int stream_index, const int step_index, InferData& infer_data)
+    const size_t thread_id, const int stream_index, const int step_index,
```

Comment:
Any other way to design this OOP so that `thread_id`, which isn't used, isn't needed as a parameter here?

Reply:
Yeah, this is certainly a code smell -- passing in an arg that isn't used. You are very observant. 

I had thought about it and I think it would require redesigning InferDataManager and/or splitting it into smaller classes so that each thread can own an instance. This would be a larger refactor and outside of the scope of this story.